spring-cloud-gcp-starters/spring-cloud-gcp-starter-config/src/main/java/org/springframework/cloud/gcp/config/GcpConfigProperties.java

```diff
@@ -41,7 +40,8 @@
 
 	private String projectId;
 
-	private Credentials credentials;
+	@NestedConfigurationProperty
```

Comment:
@joaoandremartins I thought this didn't work when you tried it before??

Reply:
That's cool! That's exactly what we have missed in the past when we tried to expose `Credentials` as top-level class.
Thank you @snicoll !

Reply:
Yes, I thought I had looked into `@NestedConfigurationProperty` before and it didn't work.
But after thinking about this for a while, I can't remember what the problem was.
It might have been because the name of the object (`Credentials`) is mapped to the name of the property.
I didn't want to call it that because end users might try to use it, but it's basically useless outside of the context of config properties. But if you're happy with this, I guess I won't oppose.